rusk-wallet/src/wallet.rs

```diff
@@ -449,22 +403,41 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
             return Err(Error::NotEnoughGas);
         }
 
+        let seed = self.store.get_seed();
+
+        let state = self.state()?;
+        let deposit = *deposit;
+
         let mut rng = StdRng::from_entropy();
-        let sender_index =
-            sender.index().expect("owned address should have an index");
+        let sender_index = sender.index()?;
+        let sender_sk = derive_phoenix_sk(seed, sender_index);
 
-        // transfer
-        let tx = wallet.execute(
+        let inputs = state
+            .inputs(sender_index, deposit + gas.limit * gas.price)?
+            .into_iter()
+            .map(|(a, b, _)| (a, b))
+            .collect();
+
+        let root = state.fetch_root()?;
+        let chain_id = state.fetch_chain_id()?;
+
+        let tx = phoenix::<_, Prover>(
             &mut rng,
-            contract_id.into(),
-            call_name,
-            call_data,
-            sender_index as u64,
-            sender.psk(),
+            &sender_sk,
+            sender.pk(),
+            reciever.pk(),
+            inputs,
+            root,
+            0,
+            true,
+            deposit,
             gas.limit,
             gas.price,
+            chain_id,
+            exec,
         )?;
-        Ok(tx)
+
+        state.prove_and_propagate(tx)
```

Comment:
Since you generated the secret key in this fn, you need to zeroize it before it goes out of scope at function return.

Reply:
If we want to zeroize (and I agree with it) this PR shouldn't store the secret key at all (right now it's stored in the Addresses struct)

And we are back to have the seed somewhere in order to derive it when needed

Reply:
zerorized